__tests__/components/TableRow.spec.ts

```diff
@@ -0,0 +1,370 @@
+            });
+
+            const vm = wrapper.vm as TableRowTs;
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(vm.isSupplyMutableMosaic).toBe(expectedResult);
+            if (expectedResult) {
+                wrapper.find('.modify-supply').trigger('click');
+                expect(wrapper.emitted('on-show-mosaic-supply-change-form')).toBeDefined();
+            } else {
+                expect(wrapper.find('.modify-supply').exists()).toBe(expectedResult);
+            }
+        };
+
+        test('returns true when mosaics is support supply mutable and not expired', () => {
+            runBasicSupplyMutableMosaic({}, true);
+        });
+
+        test('returns false when mosaics is support supply mutable is expired', () => {
+            runBasicSupplyMutableMosaic(
+                {
+                    expiration: 'expired',
+                },
+                false,
+            );
+        });
+
+        test('returns false when mosaics is not support supply mutable', () => {
+            runBasicSupplyMutableMosaic(
+                {
+                    supplyMutable: false,
+                },
+                false,
+            );
+        });
+    });
+
+    describe('hasMetadata', () => {
+        const runBasicHasMetadataTests = (params, expectedResult) => {
+            // Arrange:
+            const wrapper = getTableRowWrapper({
+                ownedAssetHexIds: [mockMosaicValue.hexId, mockNamespaceRowValue.hexId],
+                rowValues: {
+                    ...params,
+                },
+            });
+
+            const vm = wrapper.vm as TableRowTs;
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(vm.hasMetadata).toBe(expectedResult);
+
+            if (expectedResult) {
+                wrapper.find('.view-metadata').trigger('click');
+                wrapper.find('.edit-metadata').trigger('click');
+                expect(wrapper.emitted('on-show-metadata')).toBeDefined();
+                expect(wrapper.emitted('on-show-edit')).toBeDefined();
+            } else {
+                expect(wrapper.find('.view-metadata').exists()).toBe(expectedResult);
+                expect(wrapper.find('.edit-metadata').exists()).toBe(expectedResult);
+            }
+        };
+
+        test('returns true when mosaic / namespace contains metadata info', () => {
+            [mockMosaicValue, mockNamespaceRowValue].forEach((rowValue) => {
+                runBasicHasMetadataTests(
+                    {
+                        ...rowValue,
+                    },
+                    true,
+                );
+            });
+        });
+
+        test('returns false when mosaic / namespace does not contain metadata info', () => {
+            [mockMosaicValue, mockNamespaceRowValue].forEach((rowValue) => {
+                runBasicHasMetadataTests(
+                    {
+                        ...rowValue,
+                        metadataList: [],
```

Comment:
you should have another test where metadataList is undefined, since you're checking that condition in src